### PR TITLE
Cknieling/apply UI attributes on all initial shapes

### PR DESCRIPTION
**Pull request by chr11115**

### CODE REVIEW

File: src/palladio/SOPAssign.cpp [hunk 1]
```diff
@@ -1005,7 +1006,8 @@ void SOPAssign::updatePrimitiveAttributes(GU_Detail* detail) {
 								UT_String result;
 								if (choices->tokenFromIndex(result, intValue)) {
 									GA_RWHandleS stringHandle(detail->addStringTuple(attrOwner, attributeName, 1));
-									stringHandle.set(0, result);
+									for (GA_Iterator it(primitiveRange); !it.atEnd(); ++it)
+										stringHandle.set(it.getOffset(), result);
```
Comment (mistafunk) [inline src/palladio/SOPAssign.cpp:1010]: Consider deduplicating all these loops with something like this:
```
template<typename H, typename V>
void setAttributeRange(GA_Range range, H handle, V value) {
	for (GA_Iterator it(range); !it.atEnd(); ++it)
		handle.set(it.getOffset(), value);
}

```

**Review by mistafunk — CHANGES_REQUESTED** [review]
Basically all good. See inline comment regarding a de-duplication suggestion.

File: src/palladio/SOPAssign.cpp (hunk 1 above)
Comment (chr11115) [reply src/palladio/SOPAssign.cpp]: fixed

**Review by mistafunk — CHANGES_REQUESTED** [review]
Try to remove the explicit template argument, compiler should be able to deduce everything.
In any case feel free to merge afterwards.

**Review by mistafunk — APPROVED** [review]
All good.

**Pull request merged**